File: rest-common/src/main/java/org/chronopolis/rest/support/StorageUnit.java
package org.chronopolis.rest.support;

public enum StorageUnit {
    B(0), KiB(1), MiB(2), GiB(3), TiB(4), PiB(5), OOB(-1);

    private final int power;

    StorageUnit(int power) {
        this.power = power;
    }

    public StorageUnit next() {
        switch (this) {
            case B:
                return KiB;
            case KiB:
                return MiB;
            case MiB:
                return GiB;
            case GiB:
                return TiB;
            case TiB:
                return PiB;
            case PiB:
                return OOB;
            case OOB:
                return OOB;
            default:
                return OOB;
        }
    }

    public int getPower() {
        return power;
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/models/enums/StorageUnit.kt
package org.chronopolis.rest.models.enums

enum class StorageUnit(val power: Int) {
    B(0), KiB(1), MiB(2), GiB(3), TiB(4), PiB(5), OOB(-1);

    fun next() = when (this) {
        B -> KiB
        KiB -> MiB
        MiB -> GiB
        GiB -> TiB
        TiB -> PiB
        PiB -> OOB
        OOB -> OOB
    }

}